[compiler] Support operators with multiple outputs

bwasti opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #24 [WIP] Use PackedFunc to enable custom TVM operator registration from python
* #23 [benchmark] Add initial benchmark file
* #22 [compiler] Improve conversion from PyTorch IR to Relay
* #21 [testing] Improve testing to include models, image inputs and autotuned parameters
* #17 [compiler] Add None type handling with sentinel value
* **#14 [compiler] Support operators with multiple outputs**

bwasti committed:
Update on "[compiler] Support operators with multiple outputs"

[compiler] Support operators with multiple outputs

gh-metadata: pytorch tvm 14 gh/bwasti/3/head

bwasti committed:
Update on "[compiler] Support operators with multiple outputs"

[compiler] Support operators with multiple outputs

gh-metadata: pytorch tvm 14 gh/bwasti/3/head

wanchaol commented on torch_tvm/compiler.cpp:
```diff
@@ -108,24 +108,49 @@ tvm::relay::Function TVMCompiler::convertToRelay(
+      fields.push_back(value_map[sg_output]);
+    }
+    n->fields = std::move(fields);
+    return tvm::relay::Tuple(n);
```
can we actually make even the single output also fills into a Tuple, so that we could merge the single and multiple output case together? that way we could avoid checking the output size everytime.

bwasti committed:
Update on "[compiler] Support operators with multiple outputs"

[compiler] Support operators with multiple outputs

gh-metadata: pytorch tvm 14 gh/bwasti/3/head

wanchaol commented on torch_tvm/compiler.cpp:
```diff
@@ -108,24 +108,42 @@ tvm::relay::Function TVMCompiler::convertToRelay(
-        value_map[use.user->output()] = getOperator(use.user, relay_inputs);
-        new_frontier.emplace_back(use.user->output());
+        // if there are 2+ outputs, getOperator returns a tuple
+        if (use.user->outputs().size() == 1) {
```
shouldn't this if can be combine together with the else statement if in the batchnorm we eliminate the conditional on output==1?

wanchaol reviewed (approved):
LGTM, one minor suggestion

bwasti committed:
Update on "[compiler] Support operators with multiple outputs"

[compiler] Support operators with multiple outputs

gh-metadata: pytorch tvm 14 gh/bwasti/3/head
